algorithms/linfa-trees/src/decision_trees/algorithm.rs

```diff
@@ -489,6 +489,29 @@ pub struct DecisionTree<F: Float, L: Label> {
     num_features: usize,
 }
 
+impl<F: Float, L: Label + Default, D: Data<Elem = F>> PredictInplace<ArrayBase<D, Ix2>, Array2<L>>
```

Comment:
The code didn't generalise beyond Array2 until we fixed the issue where we had to have T::Owned==T. I only added this to be able to test random forest, because it didn't work for the existing Array1 implementation (since Array1::Owned==Array2). So I've just removed it entirely.